xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -409,6 +519,150 @@ void CGUIDialogVideoInfo::Update()
     pImageControl->FreeResources();
     pImageControl->SetFileName(m_movieItem->GetArt("thumb"));
   }
+  // Update the Audio Stream and Subtitles spinners
+  PLAYERCOREID eNewCore   = EPC_NONE;
+  int audioStream         = 0;
+  int subtitleStream      = 0;
+  int audioStreamCount    = 0;
+  int subtitleStreamCount = 0;
+  bool subtitlesOnFlag    = false;
+  std::vector< std::pair<std::string, int> > subsList;
+  std::vector< std::pair<std::string, int> > audioList;
+  CApplicationPlayer * m_pPlayer;
+  //FileItem list for the list selector control
+  CFileItemList* subsFileItemList  = new CFileItemList;
+  CFileItemList* audioFileItemList = new CFileItemList;
+
+  // Check if the movie we are looking at is currently playing
+  CGUIInfoBool m_isPlaying;
+  m_isPlaying.Parse("listitem.isplaying", 0);
+  m_isPlaying.Update(m_movieItem.get());
+
+  if(g_application.m_pPlayer->HasPlayer() && g_application.m_pPlayer->IsPlayingVideo() && m_isPlaying)
+  {
+    m_pPlayer = g_application.m_pPlayer;
+    // Set stream and subtitle
+    audioStream     = m_pPlayer->GetAudioStream();
+    subtitleStream  = m_pPlayer->GetSubtitle();
+    subtitlesOnFlag = m_pPlayer->GetSubtitleVisible();
+  }
+  else
+  {
+    CPlayerOptions options;
+    // If movie is not playing, Create a temporary player
+    eNewCore  = CPlayerCoreFactory::Get().GetDefaultPlayer(*m_movieItem);
+    m_pPlayer = new CApplicationPlayer;
```

Comment:
Have you looked into CDVDFileInfo::ExtractThumb? It does similar things without creating an entire player.